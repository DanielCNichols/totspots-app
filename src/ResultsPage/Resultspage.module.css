* {
  box-sizing: border-box;
}

.mobile {
  display: initial;
}

.mobileControls {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  width: 90%;
}

.mobileControls button {
  width: 100px;
}

.resultsPage {
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 70px 5px;
  margin: 0 auto;
}

.resultsControls,
.mapContainer {
  display: none;
}

@media screen and (min-width: 450px) {
  .mobile {
    display: none;
  }

  .resultsPage {
    display: grid;
    grid-template-columns: 50% 50%;
    grid-template-rows: 75px 1fr;
    padding-top: 65px;
  }

  .resultsControls {
    display: flex;
    justify-content: center;
    align-items: center;
    grid-column: span 2;
    grid-row: 1;
    width: 100%;
    /* border-right: 1px solid black; */
    text-align: center;
  }

  .resultsContainer {
    grid-column-start: auto;
    grid-row-start: 2;
    width: 100%;
    /* border-left: 1px solid black; */
    /* border-right: 1px solid black; */
    /* padding-top: 100px; */
  }

  .mapContainer {
    display: grid;
    grid-row-start: 2;
    grid-column-start: auto;
    width: 100%;
    /* border-left: 1px solid black; */
    position: sticky;
    top: 0;
    height: 100vh;
  }
}
/* @media screen and (min-width: 800px) {
  .mapContainer {
    display: none;
  }
} */
@media screen and (min-width: 1300px) {
  .mobile {
    display: none;
  }

  .resultsPage {
    display: grid;
    grid-template-columns: 20% 45% 35%;
    grid-template-rows: 1fr;
    padding-top: 50px;
    padding-bottom: 0;
  }

  .resultsControls {
    display: block;
    grid-column: 1;
    top: 0;
    width: 100%;
    height: 100vh;
    padding-top: 100px;
    position: sticky;
    /* border-right: 1px solid black; */
  }

  .resultsContainer {
    grid-column-start: 2;
    grid-row-start: 1;
    width: 100%;
    /* border-left: 1px solid black;
    border-right: 1px solid black; */
  }

  .mapContainer {
    grid-column-start: 3;
    grid-row-start: 1;
    width: 100%;
    min-width: 100%;
    /* border-left: 1px solid black; */
    position: sticky;
    padding: 100px 10px;

    top: 0;
  }

  .filterElement {
    margin-bottom: 25px;
  }
}
