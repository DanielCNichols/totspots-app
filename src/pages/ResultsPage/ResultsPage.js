import React from 'react';

export default function ResultsPage() {
  return <div>This si the results page</div>;
}
