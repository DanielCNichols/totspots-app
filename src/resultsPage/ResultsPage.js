import React from 'react'
import Result from '../result/result'
import VenueContext from '../VenuesContext'
import {Link} from 'react-router-dom'
import './Resultspage.css'


export default class ResultsPage extends React.Component {
  static contextType = VenueContext

render() {
  let {venues} = this.context
  return (
    <section className="main" role='main'>

      {/* Maybe split into another component?  */}
      <header>
        <h2>Showing results for {this.context.type} in {this.context.city}</h2>
      </header>
      <div className="addvenue">
        <p>Is something missing?</p>
        <Link to='/addVenue'><button>Suggest a new venue</button></Link>
      </div>
      <ul>
          {venues.map((venue =>(
              <Result venue={venue} key={venue.id} />
          )))}
      </ul>
      
    </section>
    )
  }
}