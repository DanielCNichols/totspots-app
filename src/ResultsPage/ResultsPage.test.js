
import React from 'react';
import ReactDOM from 'react-dom'
import ResultsPage from './ResultsPage.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'



it('renders ResultsPage without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <ResultsPage />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders ResultsPage as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <ResultsPage name="ResultsPage"/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
