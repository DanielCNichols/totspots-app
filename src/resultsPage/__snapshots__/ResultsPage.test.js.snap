// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ResultsPage as expected 1`] = `
<div>
  <div>
    <p>
      Sorry, no results found for that search
    </p>
  </div>
</div>
`;
